vault.js

```diff
@@ -1,4 +1,38 @@
 'use strict';
 module.exports = function() {
+  var val = {};
   
+  return {
+      setValue: setValue,
+      getValue: getValue
+  }
+
+  function setValue(key, value){
+    console.log('value: '+value+' key: '+key+'\n');
+    if (key !== null) {
+        val[key] = value;
+        console.log(val);
+    } else {
+        return null;
+    }
+  }
+
+  function getValue(key){
+
+      if(Object.keys(val).length > 0){
+        var keys = Object.keys(val);
+        
+        for (var i = 0; i < keys.length; i++) {
```

Comment:
Since your `val` is already an object, there is a faster way to check if a key exists, and a fast way to get that value if it does. Remember that objects are essentially hashes, or maps.